16-projects.Rmd

```diff
@@ -4,18 +4,17 @@ When you work on larger projects or reports, you may not want to put all text an
 
 ## Source external R scripts {#source-script}
 
-If your R Markdown document has a large amount of code, you may consider putting some code in external R scripts, and run these scripts via `source()`\index{source()} or `sys.source()`\index{sys.source()}, e.g.,
+If your R Markdown document has a large amount of code, you may consider putting some code in external R scripts, and running these scripts with `source()`\index{source()}[^sys-source]. This not only makes your R Markdown document cleaner, but can also make it more convenient to develop R code since (e.g.) debugging R code is often easier in pure R scripts.
 
 ````md
 ```{r, include=FALSE}`r ''`
-source("your-script.R", local = knitr::knit_global())
-# or sys.source("your-script.R", envir = knitr::knit_global())
+source("your-script.R", local = TRUE)
 ```
 ````
 
-We recommend that you use the argument `local` in `source()` or `envir` in `sys.source()` explicitly to make sure the code is evaluated in the correct environment, i.e., `knitr::knit_global()`\index{knitr!knit\_global()}. The default values for them may not be the appropriate environment: you may end up creating variables in the wrong environment, and being surprised that certain objects are not found in later code chunks.
+[^sys-source]: We don't recomend ever using `sys.source()` as it is designed specifically for internal use by base R.
```

Comment:
> We don't recommend ever using `sys.source()` as it is designed specifically for internal use by base R.

I haven't heard of this before and am surprised to learn it. Could you point out the source of this claim? The help page `?sys.source` didn't say it was designed only for internal use by base R. I mentioned `sys.source()` because it does a much simpler job than `source()`, i.e., it basically does `eval(parse())`, whereas `source()` does several other things that are irrelevant to evaluating the code.

Reply:
I think it's implied by the `envir` argument which defaults to `baseenv()` — there is no case in which a user would want this. Do you have any evidence to suggest that people are using `sys.source()` in the wild?